nativelib/src/main/scala/java/lang/ieee754tostring/ryu/RyuFloat.scala

```diff
@@ -0,0 +1,489 @@
+//
+//    3) All additional work, including introduced bugs,  is an original
+//       contribution to Scala Native development.
+
+package java.lang.ieee754tostring.ryu
+
+import java.math.BigInteger
+
+import RyuRoundingMode._
+
+object RyuFloat {
+
+  var DEBUG: Boolean = false
+
+  private val FLOAT_MANTISSA_BITS: Int = 23
+
+  private val FLOAT_MANTISSA_MASK: Int = (1 << FLOAT_MANTISSA_BITS) - 1
+
+  private val FLOAT_EXPONENT_BITS: Int = 8
+
+  private val FLOAT_EXPONENT_MASK: Int = (1 << FLOAT_EXPONENT_BITS) - 1
+
+  private val FLOAT_EXPONENT_BIAS: Int = (1 << (FLOAT_EXPONENT_BITS - 1)) - 1
+
+  private val LOG10_2_DENOMINATOR: Long = 10000000L
+
+  private val LOG10_2_NUMERATOR: Long =
+    (LOG10_2_DENOMINATOR * Math.log10(2)).toLong
+
+  private val LOG10_5_DENOMINATOR: Long = 10000000L
+
+  private val LOG10_5_NUMERATOR: Long =
+    (LOG10_5_DENOMINATOR * Math.log10(5)).toLong
+
+  private val LOG2_5_DENOMINATOR: Long = 10000000L
+
+  private val LOG2_5_NUMERATOR: Long =
+    (LOG2_5_DENOMINATOR * (Math.log(5) / Math.log(2))).toLong
+
+  private val POS_TABLE_SIZE: Int = 47
+
+  private val INV_TABLE_SIZE: Int = 31
+
+  // Only for debugging.
+  private val POW5: Array[BigInteger] = new Array[BigInteger](POS_TABLE_SIZE)
+
+  private val POW5_INV: Array[BigInteger] =
+    new Array[BigInteger](INV_TABLE_SIZE)
+
+  private val POW5_BITCOUNT: Int = 61
+
+  private val POW5_HALF_BITCOUNT: Int = 31
+
+  private val POW5_SPLIT: Array[Array[Int]] =
+    Array.ofDim[Int](POS_TABLE_SIZE, 2)
+
+  private val POW5_INV_BITCOUNT: Int = 59
+
+  private val POW5_INV_HALF_BITCOUNT: Int = 31
+
+  private val POW5_INV_SPLIT: Array[Array[Int]] =
+    Array.ofDim[Int](INV_TABLE_SIZE, 2)
+
+  val mask: BigInteger = BigInteger
+    .valueOf(1)
+    .shiftLeft(POW5_HALF_BITCOUNT)
+    .subtract(BigInteger.ONE)
+
+  val maskInv: BigInteger = BigInteger
+    .valueOf(1)
+    .shiftLeft(POW5_INV_HALF_BITCOUNT)
+    .subtract(BigInteger.ONE)
+
+  for (i <- 0 until Math.max(POW5.length, POW5_INV.length)) {
+    val pow: BigInteger       = BigInteger.valueOf(5).pow(i)
+    val pow5len: Int          = pow.bitLength()
+    val expectedPow5Bits: Int = pow5bits(i)
+    if (expectedPow5Bits != pow5len) {
+      throw new IllegalStateException(pow5len + " != " + expectedPow5Bits)
+    }
+    if (i < POW5.length) {
+      POW5(i) = pow
+    }
+    if (i < POW5_SPLIT.length) {
+      POW5_SPLIT(i)(0) = pow
+        .shiftRight(pow5len - POW5_BITCOUNT + POW5_HALF_BITCOUNT)
+        .intValue()
+      POW5_SPLIT(i)(1) =
+        pow.shiftRight(pow5len - POW5_BITCOUNT).and(mask).intValue()
+    }
+    if (i < POW5_INV.length) {
+      val j: Int = pow5len - 1 + POW5_INV_BITCOUNT
+      val inv: BigInteger =
+        BigInteger.ONE.shiftLeft(j).divide(pow).add(BigInteger.ONE)
+      POW5_INV(i) = inv
+      POW5_INV_SPLIT(i)(0) = inv.shiftRight(POW5_INV_HALF_BITCOUNT).intValue()
+      POW5_INV_SPLIT(i)(1) = inv.and(maskInv).intValue()
+    }
+  }
```

Comment:
Can we codegen the `POW5`, `POW5_INV` and `POW_INV_SPLIT`? This would remove the dependency on `BigInteger` here.

Reply:
Done & committed for RyuDouble.scala.  `test-all` passes.  
RyuFloat.scala still needs corresponding changes. I want the
changes in RyuDouble.scala to pass review (or be declined)
before I propagate them.  Progress....

Reply:
Lets do the same for `RyuFloat` now. The only change I would make is to use one dimensional arrays.

Reply:
Done!